src/rpc/methods/gas.rs

```diff
@@ -164,13 +158,41 @@ pub async fn estimate_gas_premium<DB: Blockstore>(
         .unwrap()
         .sample(&mut crate::utils::rand::forest_rng());
 
-    premium *= BigInt::from_f64(noise * (1i64 << precision) as f64)
+    premium *= BigInt::from_f64((noise * (1i64 << precision) as f64) + 1f64)
         .context("failed to convert gas premium f64 to bigint")?;
     premium = premium.div_floor(1i64 << precision);
 
     Ok(premium)
 }
 
+// finds 55th percentile instead of median to put negative pressure on gas price
+fn median_gas_premium_calculation(mut prices: Vec<GasMeta>, blocks: u64) -> TokenAmount {
+    prices.sort_by(|a, b| b.price.cmp(&a.price));
+
+    let mut at = BLOCK_GAS_TARGET * blocks / 2;
+    at += BLOCK_GAS_TARGET * blocks / (2 * 20);
```

Comment:
why `2*20`? Why exactly 55th percentile? Can we use a dedicated crate for computing percentiles?

Reply:
Let's investigate this and ensure the implementation matches assumptions and we understand the reasoning behind magic numbers.